nwbwriter.py

```diff
@@ -118,49 +167,54 @@ def save_NWB(CIDataContainerObj,filename,imaging_plane_name=None,imaging_series_
+            # ps.add_column('rejected', 'in rejected list')
 
             # Add ROIs
-            if not hasattr(CIDataContainerObj, 'accepted_list'):
-                for i, (roi, snr, r, cnn) in enumerate(zip(CIDataContainerObj.A.T, CIDataContainerObj.SNR_comp, CIDataContainerObj.r_values, CIDataContainerObj.cnn_preds)):
-                    ps.add_roi(image_mask=roi.T.reshape(CIDataContainerObj.dims), r=r, snr=snr, cnn=cnn,
-                               keep=i in CIDataContainerObj.idx_components, accepted=False, rejected=False)
+            if not hasattr(TraceExtractor, 'accepted_list'):
+                for i, (roi, snr, r, cnn) in enumerate(zip(TraceExtractor.Masks.T, TraceExtractor.SNR_comp, TraceExtractor.r_values, TraceExtractor.cnn_preds)):
+                    ps.add_roi(image_mask=roi.T.reshape(TraceExtractor.dims), r=r, snr=snr, cnn=cnn,
+                               keep=i in TraceExtractor.idx_components, accepted=False, rejected=False)
             else:
-                for i, (roi, snr, r, cnn) in enumerate(zip(CIDataContainerObj.A.T, CIDataContainerObj.SNR_comp, CIDataContainerObj.r_values, CIDataContainerObj.cnn_preds)):
-                    ps.add_roi(image_mask=roi.T.reshape(CIDataContainerObj.dims), r=r, snr=snr, cnn=cnn,
-                               keep=i in CIDataContainerObj.idx_components, accepted=i in CIDataContainerObj.accepted_list, rejected=i in CIDataContainerObj.rejected_list)
+                for i, (roi, snr, r, cnn) in enumerate(zip(TraceExtractor.Masks.T, TraceExtractor.SNR_comp, TraceExtractor.r_values, TraceExtractor.cnn_preds)):
+                    ps.add_roi(image_mask=roi.T.reshape(TraceExtractor.dims), r=r, snr=snr, cnn=cnn,
+                               keep=i in TraceExtractor.idx_components, accepted=i in TraceExtractor.accepted_list, rejected=i in TraceExtractor.rejected_list)
 
-            for bg in CIDataContainerObj.b.T:  # Backgrounds
-                ps.add_roi(image_mask=bg.reshape(CIDataContainerObj.dims), r=np.nan, snr=np.nan, cnn=np.nan, keep=False, accepted=False, rejected=False)
+            for bg in TraceExtractor.Masks_b.T:  # Backgrounds
+                ps.add_roi(image_mask=bg.reshape(TraceExtractor.dims), r=np.nan, snr=np.nan, cnn=np.nan, keep=False, accepted=False, rejected=False)
             # Add Traces
-            n_rois = CIDataContainerObj.A.shape[-1]
-            n_bg = len(CIDataContainerObj.f)
+            n_rois = TraceExtractor.Masks.shape[-1]
+            n_bg = len(TraceExtractor.Roi_response_b)
             rt_region_roi = ps.create_roi_table_region(
                 'ROIs', region=list(range(n_rois)))
 
             rt_region_bg = ps.create_roi_table_region(
                 'Background', region=list(range(n_rois, n_rois+n_bg)))
 
-            timestamps = np.arange(CIDataContainerObj.f.shape[1]) / imaging_rate + starting_time
+            timestamps = np.arange(TraceExtractor.Roi_response_b.shape[1]) / imaging_rate + starting_time
```

Comment:
TimeSeries objects can be defined by rate and starting_time instead of timestamps, and should be defined this way if the sampling rate is constant as it is here